### PR TITLE
S1AP update and Bug fixes

### PR DESCRIPTION
**Pull request by badhrinathpa**
1. S1AP encoder/decoder header files and library added to MME. MME code updated to use library to decode all the messages it currently handles.
2. Added code in MME to handle Authentication Synch failure. The MME sends the AUTS received in the Synch failure and sends AIR again to HSS. HSS sends new set of vectors and MME sends authentication request again with new set of vectors. This functionality has a corresponding update on c3po repo also.
3. Bug fix to send proper Apn name in Initial context setup request ESM container.
4. SCTP stream Id updated with temporary fix to use different stream Id in case of UE related messages compared to setup messages.

### CODE REVIEW

**Comment by thakurajayL** [comment]
src/s1ap/asn1c/asn1c   this  is binary file. We want to commit it ?

File: src/s1ap/handlers/ctxrelease_resp.c [hunk 1]
```diff
@@ -36,22 +36,16 @@ extern int g_enb_fd;
...
+s1_ctx_release_resp_handler(InitiatingMessage_t *msg)
 {
 	struct ctx_release_complete_Q_msg release_complete;
 	struct proto_IE s1_ctx_release_ies;
 
-	unsigned short msg_len = get_length(&msg);
-
-	char *buffer;
-	log_msg(LOG_INFO, "S1AP_UE_CTX_RELEASE msg: %s\n", msg_to_hex_str(msg, msg_len, &buffer));
-	free(buffer);
```
Comment (thakurajayL) [inline src/s1ap/handlers/ctxrelease_resp.c:48]: Did we removed this logging ?

Comment (badhrinathpa) [reply src/s1ap/handlers/ctxrelease_resp.c:48]: yes. Now the decoder already gives us a structure variable. So this character buffer is not available anymore.

**Review by thakurajayL — APPROVED** [review]
May be a note is required about this whole code is generated will be useful.
e.g. feeding message format from 3gpp standard==>this generates .h files and .a static lib ==>In main s1ap Makefile we link .a library and wherever required we include corresponding .h files.

**Pull request merged**